src/sidebar/groups.js

```diff
@@ -74,29 +118,45 @@ function groups(localStorage, serviceUrl, session, $rootScope, store) {
     var prevFocused = focused();
     var g = get(id);
     if (g) {
-      focusedGroup = g;
+      focusedGroupId = g.id;
       localStorage.setItem(STORAGE_KEY, g.id);
       if (prevFocused.id !== g.id) {
         $rootScope.$broadcast(events.GROUP_FOCUSED, g.id);
       }
     }
   }
-
+  
   // reset the focused group if the user leaves it
   $rootScope.$on(events.GROUPS_CHANGED, function () {
-    if (focusedGroup) {
-      focusedGroup = get(focusedGroup.id);
-      if (!focusedGroup) {
-        $rootScope.$broadcast(events.GROUP_FOCUSED, focused());
+    // return for use in test
+    return load();
+  });
+
+  // refetch the list of groups when user changes
+  $rootScope.$on(events.USER_CHANGED, () => {
+    // FIXME Makes a second api call on page load. better way?
+    // return for use in test
+    return load();
```

Comment:
I only saw one call to `/api/groups` on page load. I haven't looked into why yet.